OS/DiskOS/terminal.lua

```diff
@@ -389,17 +401,31 @@ function term.loop() --Enter the while loop of the terminal
           blink = true; checkCursor()
         end
       elseif a == "left" then
-        blink = false; checkCursor()
-        if inputPos > 1 then
-          inputPos = inputPos - 1
-          printBackspace(-1)
+        --If an autocomplete suggestion is displayed, erase it
+        if next(autocompleteSuggestions) ~= nil then
+          term.clearSuggestion(autocompleteSuggestions[suggestionIndex
+        ])
+          autocompleteSuggestions = {}
+        else
+          blink = false; checkCursor()
+          if inputPos > 1 then
+            inputPos = inputPos - 1
+            printBackspace(-1)
+          end
+          blink = true; checkCursor()
         end
-        blink = true; checkCursor()
       elseif a == "right" then
-        blink = false; checkCursor()
-        if inputPos <= buffer:len() then
-          print(buffer:sub(inputPos,inputPos),false)
-          inputPos = inputPos + 1
+        --If an autocomplete suggestion is displayed, accept it
+        if next(autocompleteSuggestions) ~= nil then
```

Comment:
This expression could be simplified:
 Values in acts as booleans:
- false, nil -> false.
- true, number, table, string, userdata -> true.

So this expression could be written as:
```lua
if next(autocompleteSuggestions) then
```